timemachine/fe/rbfe.py

```diff
@@ -400,6 +445,130 @@ def estimate_relative_free_energy(
         raise err
 
 
+def estimate_relative_free_energy_via_greedy_bisection(
+    mol_a,
+    mol_b,
+    core: NDArray,
+    ff: Forcefield,
+    host_config: Optional[HostConfig],
+    seed: int,
+    n_frames: int = 1000,
+    prefix: str = "",
+    lambda_interval: Optional[Tuple[float, float]] = None,
+    n_windows: Optional[int] = None,
+    n_eq_steps: int = 10000,
+    steps_per_frame: int = 400,
+    min_cutoff: float = 0.7,
+) -> SimulationResult:
+    r"""Estimate relative free energy between mol_a and mol_b via independent simulations with a dynamic lambda schedule
+    determined by successively bisecting the lambda interval between the pair of states with the greatest BAR
+    :math:`\Delta G` error. Molecules should be aligned to each other and within the host environment.
+
+    Parameters
+    ----------
+    mol_a: Chem.Mol
+        initial molecule
+
+    mol_b: Chem.Mol
+        target molecule
+
+    core: list of 2-tuples
+        atom_mapping of atoms in mol_a into atoms in mol_b
+
+    ff: Forcefield
+        Forcefield to be used for the system
+
+    host_config: HostConfig or None
+        Configuration for the host system. If None, then the vacuum leg is run.
+
+    n_frames: int
+        number of samples to generate for each lambda window, where each sample is `steps_per_frame` steps of MD.
+
+    prefix: str
+        A prefix to append to figures
+
+    seed: int
+        Random seed to use for the simulations.
+
+    lambda_interval: (float, float) or None, optional
+        Minimum and maximum value of lambda for the transformation; typically (0, 1), but sometimes useful to choose
+        other values for testing.
+
+    n_windows: int or None, optional
+        Number of windows used for interpolating the lambda schedule with additional windows. Additionally controls
+        the number of evenly-spaced lambda windows used for initial conformer optimization. Defaults to 30 windows.
+
+    n_eq_steps: int
+        Number of equilibration steps for each window.
+
+    steps_per_frame: int
+        The number of steps to take before collecting a frame
+
+    min_cutoff: float
+        throw error if any atom moves more than this distance (nm) after minimization
+
+    Returns
+    -------
+    SimulationResult
+        Collected data from the simulation (see class for storage information). Returned frames and boxes
+        are defined by keep_idxs.
+    """
+
+    assert n_windows is None or n_windows >= 2
+
+    single_topology = SingleTopology(mol_a, mol_b, core, ff)
+
+    lambda_min, lambda_max = lambda_interval or (0.0, 1.0)
+    lambda_grid = np.linspace(lambda_min, lambda_max, n_windows or 30)
```

Comment:
nit: Add constant for DEFAULT_NUM_WINDOWS?

Reply:
Added in https://github.com/proteneer/timemachine/pull/959/commits/5737c4473ea996e4e2f7b311372737ee1e6bef41